packages/block-library/src/image/edit.native.js

```diff
@@ -1,68 +1,131 @@
 /**
  * External dependencies
  */
-import { View, Image, TextInput } from 'react-native';
-import RNReactNativeGutenbergBridge from 'react-native-gutenberg-bridge';
+import React from 'react';
+import { View, Image, TextInput, NativeEventEmitter } from 'react-native';
+import {
+	subscribeMediaUpload,
+	onMediaLibraryPressed,
+	onUploadMediaPressed,
+} from 'react-native-gutenberg-bridge';
 
 /**
  * Internal dependencies
  */
 import { MediaPlaceholder, RichText, BlockControls } from '@wordpress/editor';
-import { Toolbar, ToolbarButton } from '@wordpress/components';
+import { Toolbar, ToolbarButton, Spinner } from '@wordpress/components';
 import { Component } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import ImageSize from './image-size';
 
-class ImageEdit extends Component {
-	constructor() {
-		super( ...arguments );
-		this.onMediaLibraryPress = this.onMediaLibraryPress.bind( this );
+const MEDIA_ULOAD_STATE_UPLOADING = 1;
+const MEDIA_ULOAD_STATE_SUCCEEDED = 2;
+const MEDIA_ULOAD_STATE_FAILED = 3;
+
+export default class ImageEdit extends React.Component {
+
+	subscriptionParentMediaUpload: ?EmitterSubscription;
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			progress: 0,
+			isUploadInProgress: false,
+		};
+
+		this.mediaUpload = this.mediaUpload.bind( this );
+		this.addMediaUploadListener = this.addMediaUploadListener.bind( this );
+		this.removeMediaUploadListener = this.removeMediaUploadListener.bind( this );
+		this.finishMediaUploading = this.finishMediaUploading.bind( this );
 	}
 
-	onUploadPress() {
-		// This method should present an image picker from
-		// the device.
-		//TODO: Implement upload image method.
+	componentWillUnmount() {
+		this.removeMediaUploadListener();
+	}
+
+	mediaUpload( payload ) {
+		const { attributes } = this.props;
+
+		if (payload.mediaId === attributes.id) {
+			if ( payload.state === MEDIA_ULOAD_STATE_UPLOADING ) {
+				this.setState( { progress: payload.progress, isUploadInProgress: true } );
+			} else if ( payload.state === MEDIA_ULOAD_STATE_SUCCEEDED || payload.state === MEDIA_ULOAD_STATE_FAILED ) {
```

Comment:
At the moment the success and failed state are displaying the same end result: they are assuming success.

Should the failed state be implemented on this PR?